components/mhz19b/mhz19b.c

```diff
@@ -0,0 +1,254 @@
+/**
+ * @file mhz19b.c
+ *
+ * ESP-IDF driver for MH-Z19B NDIR CO2 sensor connected to UART
+ *
+ * Inspired from https://github.com/Erriez/ErriezMHZ19B
+ *
+ * Copyright (C) 2021 David Douard <david.douard@sdfa3.org>
+ *
+ * BSD Licensed as described in the file LICENSE
+ */
+#include <string.h>
+#include <esp_idf_lib_helpers.h>
+#include <esp_log.h>
+#include <esp_timer.h>.
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "mhz19b.h"
+
+static const char *TAG = "mhz19b";
+
+esp_err_t mhz19b_init(mhz19b_dev_t *dev, uart_port_t uart_port, gpio_num_t tx_gpio, gpio_num_t rx_gpio)
+{
+    uart_config_t uart_config = {
+        .baud_rate = 9600,
+        .data_bits = UART_DATA_8_BITS,
+        .parity    = UART_PARITY_DISABLE,
+        .stop_bits = UART_STOP_BITS_1,
+        .flow_ctrl = UART_HW_FLOWCTRL_DISABLE,
+        .source_clk = UART_SCLK_APB,
+    };
+    uart_driver_install(uart_port, MHZ19B_SERIAL_BUF_LEN * 2, 0, 0, NULL, 0);
```

Comment:
`uart_driver_install()` returns esp_err_t. Please check it and return error if any (use `CHECK()` macro as in other drivers). Same goes for the rest of the function calls.